x/archivedb/key.go

```diff
@@ -0,0 +1,76 @@
+package archivedb
+
+import (
+	"errors"
+	"math"
+)
+
+var errBadVersion = errors.New("invalid version")
+
+// Key
+//
+// The keys contains a few extra information alongside the given key. The key is
+// what is known outside of this internal scope as the key, but this struct has
+// more information compacted inside the Key to take advantage of natural
+// sorting. The inversed height is stored right after the Prefix, which is
+// MaxUint64 minus the desired Height. An extra byte is also packed inside the
+// key to indicate if the key is a deletion of any previous value (because
+// archivedb is an append only database).
+//
+// Any other property are to not serialized but they are useful when parsing a
+// Key struct from the database
+type Key struct {
+	Prefix         []byte `serialize:"true"`
+	HeightReversed uint64 `serialize:"true"`
+	IsDeleted      bool   `serialize:"true"`
```

Comment:
Is the rational for storing this in the key rather than the value to avoid a mem copy? I would have intuitively thought this would make more sense as part of the value

Reply:
Probably, I'll think about it and I'll test both approaches inside benchmark script that I described above that I will do